test/commands/user_track/generate_concepts_status_test.rb

```diff
@@ -0,0 +1,201 @@
+require "test_helper"
+
+class UserTrack::GenerateConceptsStatusTest < ActiveSupport::TestCase
+  test "generates concepts for empty track" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    assert_equal(
+      {},
+      ::UserTrack::GenerateConceptsStatus.(user_track)
+    )
+  end
+
+  test "concepts without prerequisite are :unlocked" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    create :track_concept, slug: 'basics', track: track
+
+    assert_equal(
+      { 'basics' => :unlocked },
+      ::UserTrack::GenerateConceptsStatus.(user_track)
+    )
+  end
+
+  test "concepts following incomplete concepts are :locked" do
+    track = create :track
+    user_track = create :user_track, track: track
```

Comment:
This is a good example of where I'd definitely define `track = ...` at the top, because its used so often and is more the core model in this test. (So leave this exactly as is)